scripts/s3uploading.js

```diff
@@ -0,0 +1,149 @@
+     shell.exec(command, (exitCode, stdout, stderr) => (exitCode !== 0) ? reject(new Error(stderr)) : resolve(stdout))
+);
+
+const checkout = async (module, moduleName) => {
+    /**
+     * clone each repository to an individual directory based on sha1(moduleName)
+     * we can have repository with the same name but from different git hosting services
+     */
+    const moduleLocalPath = `${tmp}/${stringToSha1(moduleName)}`;
+    // GIT_TERMINAL_PROMPT=0 disables git clone prompting for credentials in case of private or non-existing repository
+    await exec(`GIT_TERMINAL_PROMPT=0 git clone --no-checkout ${module.repo} ${moduleLocalPath}`)
+        .catch(e => exit(`Error occurred while ${module.repo} cloning. Err. ${e.message}`));
+    shell.cd(moduleLocalPath);
+
+    await exec(`git checkout ${module.commit}`)
+        .catch(e => exit(`Error occurred while ${module.repo} ${module.commit} checkout. Err. ${e.message}`));
+
+    if (module.subdirectory && module.subdirectory.length > 0) {
+        shell.cd(module.subdirectory);
+    }
+}
+
+const createHashFromFile = filePath => new Promise(resolve => {
+    const hash = crypto.createHash('sha256');
+    fs.createReadStream(filePath).on('data', data => hash.update(data)).on('end', () => resolve(hash.digest('hex')));
+});
+
+const stringToSha1 = string => crypto.createHash('sha1').update(string).digest('hex');
+
+const processArchive = async (name, module) => {
+    const archiveBaseName  = `${module.commit}.tar.gz`;
+    await exec(`git archive ${module.commit} --format tar.gz --output ${archiveBaseName}`)
+        .catch(e => exit(`Error occurred while archiving ${name} module. Err. ${e.message}`));
+    let hash = await createHashFromFile(`./${archiveBaseName}`);
+    const s3path = `modules/${name}/${module.commit}.tar.gz`;
+    uploadQueue.push({localPath: `${shell.pwd().toString()}/${module.commit}.tar.gz`, s3path});
+    return {"archive_url": s3path, "archive_sha256": hash}
+}
+
+const processReadme = async (moduleName, module) => {
+    let readme_url = null, readme_sha256 = null;
+    for (const file of shell.ls('*')) {
+        if (readmeRegex.test(file)) {
+            readme_url = `modules/${moduleName}/${module.commit}${path.extname(file)}`;
+            uploadQueue.push({localPath: `${shell.pwd().toString()}/${file}`, s3path: readme_url});
+            readme_sha256 = await createHashFromFile(`./${file}`);
+        }
+    }
+    return {readme_url, readme_sha256};
+}
+
+const processModules = async () => {
+    for (const moduleName in modules) {
+        const module = modules[moduleName];
+        // skip if alias or version already exists
+        if (
+            module.hasOwnProperty('alias') ||
+            (versions.hasOwnProperty(moduleName) && versions[moduleName].hasOwnProperty(module.version))
+        ) continue;
+
+        if (!module.commit || module.commit.length == 0) {
+            exit(`${moduleName} module does not have commit`);
```

Comment:
why `exit()` and `continue`? This doesn't seem to agree. Shouldn't it be like `warn()` or `console.error()` and `continue`?